pkg/infra/tower.go

```diff
@@ -66,7 +65,8 @@ func (t *Tower) Tick() {
 	}
 	// Decrease agent HP and reset tower at end of day
 	if t.dayInfo.CurrTick%t.dayInfo.TicksPerDay == 0 {
-		t.hpDecay() // decreases HP and kills if < 0
+		t.endOfDay()
```

Comment:
Hmm I'd say let's have the treaty part in its own PR once the implementation is done and keep this nice `t.endOfDay()` change?